docs/developer_overview.md

```diff
@@ -0,0 +1,527 @@
+# CUB Developer Overview
```

Comment:
I would encourage the use of [semantic line breaks](https://ramshankar.org/blog/posts/2019/semantic-line-breaks/) in this document. Some lines are very long and contain multiple sentences. It can be easier to edit and diff Markdown documents when each sentence is written on a separate line. We have adopted this for RAPIDS cuDF documentation and it has been helpful for writing, editing, and reviewing.

Reply:
Thanks, I didn't know about that approach. It makes sense for version control.